qiskit/aqua/components/neural_networks/quantum_generator.py

```diff
@@ -62,6 +64,7 @@ def __init__(self,
                 or a QuantumCircuit implementing the generator.
             init_params: 1D numpy array or list, Initialization for
                 the generator's parameters.
+            optimizer: optimizer to be used for the training of the generator
```

Comment:
Are there any specific aspects or configuration of the optimizer that we could add here to the docstring to inform the user? For instance I see that the default optimizer just has one iteration.

Also in train I see this code that I think will fail if another optimizer is supplied.
```
        self._optimizer._maxiter = 1
        self._optimizer._t = 0
```

Could we have a test case with different optimizer to assure things work.